metrics/metrics.py

```diff
@@ -0,0 +1,45 @@
+class Metrics(object):
+    """Metrics abstract layer, accept uri-style configuration
+    """
+    def __init__(self, metrics_uri):
+        import urlparse
+
+        # parse the uri
+        parsed_metrics_url = urlparse.urlsplit(metrics_uri)
+
+        # validate metrics uri is set with correct scheme, `metrics`
+        metrics_url_scheme = parsed_metrics_url.scheme
+        if metrics_url_scheme != 'metrics':
+            raise ValueError('Metrics URI must start with metrics://, got %s' % metrics_url_scheme)
+
+        # api key has to be in the auth's user name
+        api_key = parsed_metrics_url.username
+        parsed_options = urlparse.parse_qs(parsed_metrics_url.query)
+
+        # parsed options is a dictionary with querystring
+        # value as list (since it's legal to contain duplicate keys),
+        # just take the first value
+        options = {k:v_list[0] for k, v_list in parsed_options.iteritems()}
+
+        backend_cls = self._load_backend_module(parsed_metrics_url.hostname)
+
+        self.backend = backend_cls(api_key=api_key, **options)
+
+    def _load_backend_module(self, backend_name):
+
+        from . import backends
```

Comment:
Doesn't lint complain about relative imports? Is there any reason this can't be an absolute import instead? Would rather add the future absolute_import bit if we need it.